statistics/handle.go

```diff
@@ -156,3 +156,30 @@ func (h *Handle) UpdateTableStats(tables []*Table, deletedIDs []int64) {
 	}
 	h.statsCache.Store(newCache)
 }
+
+// LoadNeededHistograms will load histograms for those needed columns.
+func (h *Handle) LoadNeededHistograms() error {
+	neededColumns.m.Lock()
+	colsMap := make(map[int64][]*Column)
+	for id, cols := range neededColumns.cols {
+		colsMap[id] = cols
+		delete(neededColumns.cols, id)
```

Comment:
I think it's better to only read the map first, then delete the map after will have loaded the histograms.